File: android/app/src/main/java/com/shenxian/promotion/core/GameState.kt
package com.shenxian.promotion

class GameState {
    var day: Int = 1
    var merit: Int = 0
    var incense: Int = 0
    var position: String = "无品神籍"
    var identity: String = "凡人"
    var reputation: Int = 0

    fun summary(): String {
        return "身份：$identity\n职位：$position\n日期：第${day}日\n功德：$merit\n香火：$incense\n声望：$reputation"
    }
}

File: src/core/game-state.ts
export interface PlayerState {
  id: string;
  name: string;
  realm: string;
  rank: string;
  merit: number;
  reputation: number;
}

export interface GameState {
  version: string;
  day: number;
  year: number;
  player: PlayerState;
}

export function createDefaultGameState(): GameState {
  return {
    version: "0.1.0",
    day: 1,
    year: 1,
    player: {
      id: "player",
      name: "未命名修行者",
      realm: "凡人",
      rank: "无品",
      merit: 0,
      reputation: 0
    }
  };
}
